doc/ChangeLog

```diff
@@ -1,4 +1,100 @@
+Date: Fri 31 May 2024 11:24:29 AM MDT
+One-line Summary: Rm manage_externals and update documentation accordingly
+
+Purpose and description of changes
+----------------------------------
+#2443 Jim replaced this PR with the next one
+#2559 Remove manage externals
+#2564 Replace checkout_externals with git-fleximod in documentation
+
+Significant changes to scientifically-supported configurations
+--------------------------------------------------------------
+
+Does this tag change answers significantly for any of the following physics configurations?
+(Details of any changes will be given in the "Answer changes" section below.)
+
+    [Put an [X] in the box for any configuration with significant answer changes.]
+
+[ ] clm6_0
+
+[ ] clm5_1
+
+[ ] clm5_0
+
+[ ] ctsm5_0-nwp
+
+[ ] clm4_5
+
+
+Bugs fixed
+----------
+List of CTSM issues fixed (include CTSM Issue # and description) [one per line]:
+Fixes #2537 Remove manage_externals stuff from run_sys_tests
+Fixes #2536 Update documentation for git-fleximod
+Fixes #2467 Remove references to cheyenne from the testlist
+
+Notes of particular relevance for users
+---------------------------------------
+Caveats for users (e.g., need to interpolate initial conditions):
+ ./manage_externals/checkout_externals
+ is replaced with
+ ./bin/git-fleximod
+
+Changes to documentation:
+ Accordingly
+
+Testing summary:
+----------------
+
+ [PASS means all tests PASS; OK means tests PASS other than expected fails.]
+
+  build-namelist tests (if CLMBuildNamelist.pm has changed):
+
+    derecho - PASS
+
+  python testing (if python code has changed; see instructions in python/README.md; document testing done):
+
+    derecho - PASS
+
+    [If python code has changed and you are NOT running aux_clm (e.g., because the only changes are in python
+     code) then also run the clm_pymods test suite; this is a small subset of aux_clm that runs the system
+     tests impacted by python changes. The best way to do this, if you expect no changes from the last tag in
+     either model output or namelists, is: create sym links pointing to the last tag's baseline directory,
+     named with the upcoming tag; then run the clm_pymods test suite comparing against these baselines but NOT
+     doing their own baseline generation. If you are already running the full aux_clm then you do NOT need to
+     separately run the clm_pymods test suite, and you can remove the following line.]
+
+    clm_pymods test suite on derecho - 
+
+  regular tests (aux_clm: https://github.com/ESCOMP/CTSM/wiki/System-Testing-Guide#pre-merge-system-testing):
+
+    derecho ----- OK
+    izumi ------- OK
+
+  any other testing (give details below):
+
+  ctsm_sci
+     derecho ---- OK
+
+Answer changes
+--------------
+Changes answers relative to baseline: No
+
+Other details
+-------------
+List any externals directories updated (cime, rtm, mosart, cism, fates, etc.):
+ JIM: What's the answer here?
```

Comment:
@jedwards4b do we need to include anything here for this tag?

Reply:
@slevis-lmwg we should update the word "externals" here to "git submodules", and also do that in the template file as well.

Verify that the submodules are all the same versions from the previous tag. They should be, but we should verify. Trust -- but verify.

Reply:
Diffs in externals that I see between this branch and ctsm5.2.006:
cismwrap_2_2_001 vs. cismwrap_2_1_100
cime6.0.246 vs. cime6.0.238_httpsbranch01
cdeps1.0.34 vs. cdeps1.0.33

Let me know how we should proceed. Meanwhile I will make the updates that you pointed out in your last post here.

Reply:
OK, the cismwrap and cdeps differences are just due to git-fleximod. The cime differences are larger, but your testing shows that it doesn't change answers. So it's all good.

So document those in the ChangeLog.

You might confirm that your testing was done with those versions. The only changes since the testing was sent are:

```
 .lib/git-fleximod/License                                                       |   2 +-
 .lib/git-fleximod/README.md                                                     |   2 -
 .lib/git-fleximod/git_fleximod/cli.py                                           |  26 +++--
 .lib/git-fleximod/git_fleximod/git_fleximod.py                                  |  10 +-
 .lib/git-fleximod/pyproject.toml                                                |   2 +-
 .lib/git-fleximod/tbump.toml                                                    |   2 +-
 README                                                                          |   1 +
 README_EXTERNALS.rst                                                            | 132 ----------------------
 README_GITFLEXIMOD.rst                                                          | 115 ++++++++++++++++++++
 bld/CLMBuildNamelist.pm                                                         |   2 +-
 cime                                                                            |   0
 cime_config/testdefs/testlist_clm.xml                                           | 651 +------------------------------------------------------------------------------------------------------------
 doc/README.CHECKLIST.master_tags                                                |   6 +-
 doc/build_docs                                                                  |   2 +-
 doc/design/python_script_user_interface.rst                                     |   2 +-
 doc/source/lilac/obtaining-building-and-running/obtaining-and-building-ctsm.rst |  25 +++--
 doc/source/lilac/specific-atm-models/wrf.rst                                    |   2 +-
 manage_externals/checkout_externals                                             |   3 +
 python/ctsm/site_and_regional/run_neon.py                                       |   2 +-
 tools/mksurfdata_esmf/README.md                                                 |   2 +-
```

So the only code touched was git-fleximod, testlist_clm.xml (to remove cheyenne), and a comment in run_neon.py. So none of those seem to warrant redoing testing to me. I ran query_testlists to make sure the testlist XML file doesn't have syntax errors and that worked.